src/provider/polling-util.ts

```diff
@@ -30,7 +30,7 @@ export const createPollingJsonRpcProviderForListeners = async (
     return provider;
   }
 
-  if ('pollingInterval' in provider) {
+  if (provider instanceof JsonRpcProvider) {
```

Comment:
This is because our fork is based on Ethers 6.7.x which changed some fields in providers, such that `AbstractProvider` now has `pollingInterval`, which in my understanding means that all providers have that field.